portal/static/js/utility.js

```diff
@@ -45,59 +17,57 @@ function showMain() {
         "-khtml-opacity": 1,
         "opacity": 1
     });
-}
-function hideLoader(delay, time) {
+};
+UtilityObj.prototype.hideLoader = function(delay, time) {
     if (delay) {
         $("#loadingIndicator").hide();
-    } else {
-        setTimeout(function() {
-            $("#loadingIndicator").fadeOut();
-        }, time || 200);
+        return;
     }
-}
-// Loading indicator that appears in UI on page loads and when saving
-function loader(show) {
+    setTimeout(function() {
+        $("#loadingIndicator").fadeOut();
+    }, time || 200);
+};
+UtilityObj.prototype.loader = function(show) {
     //landing page
-    if ($("#fullSizeContainer").length > 0) {
-        hideLoader();
-        showMain();
+    if (document.getElementById("fullSizeContainer")) {
+        this.hideLoader();
+        this.showMain();
         return false;
     }
     if (show) {
         $("#loadingIndicator").show();
-    } else {
-        if ((typeof DELAY_LOADING !== "undefined") && !DELAY_LOADING) { /*global DELAY_LOADING*/
-            setTimeout(function() {
-                showMain();
-            }, 100);
-            hideLoader(true, 350);
-        }
+        return;
+    }
+    if (!this.isDelayLoading()) {
+        var self = this;
+        setTimeout(function() {
+            self.showMain();
+        }, 100);
+        this.hideLoader(true, 350);
     }
-}
-function _isTouchDevice() {
+};
+UtilityObj.prototype.isDelayLoading = function() { /*global DELAY_LOADING*/
+    return (typeof DELAY_LOADING !== "undefined") && DELAY_LOADING;
+};
+UtilityObj.prototype.isTouchDevice = function() {
```

Comment:
Is there a way to move this kind of differentiation lower in the code? I worry it highlights the differentiation between devices, instead of smoothing it over.

Ideally we would have a cohesive and singular view of handling any browser client, only handling browser idiosyncrasies as necessary.